tests/integration/testlib/ifacelib.py

```diff
@@ -54,3 +54,11 @@ def _set_eth_admin_state(ifname, state):
             ]
         }
         libnmstate.apply(desired_state)
+
+
+def get_mac_address(ifname):
+    state = statelib.show_only((ifname,))
```

Comment:
Since you are using `show_only()` the for loop for the interfaces is not needed. IMHO it is a little bit irritating that this will implicitly get the current state instead of accepting a state to get the mac address from. Maybe call it `get_current_mac_address()` to tell this.